src/meta-srv/src/metasrv.rs

```diff
@@ -339,6 +358,8 @@ impl MetaSrv {
         let mailbox = self.mailbox.clone();
         let election = self.election.clone();
         let skip_all = Arc::new(AtomicBool::new(false));
+        let publish = self.publish.clone();
```

Comment:
Suggest placing the publish object into the reporterHandler (when constructing the reporterHandler) for now, and decide whether to put it into ctx in the future according to usage.

Reply:
Good suggestion!